v2/oauth/machine-to-machine-authentication.mdx

```diff
@@ -0,0 +1,123 @@
+:::
+
+
+## 1. Create the OAuth2 Clients
+
+For each of your **`microservices`** you will have to create a separate [**OAuth2 client**](/docs/oauth/introduction#client).
+This can be done by directly calling the **SuperTokens Core** API.
+
+<CoreInjector defaultValue="https://example.com">
+
+```bash
+curl -X POST ^{coreInjector_uri_without_quotes}/recipe/oauth2/admin/clients \
+  -H "Content-Type: application/json" \
+  -H "api-key: ^{coreInjector_api_key_without_quotes}" \
+  -d '{
+      "client_name": "<YOUR_CLIENT_NAME>",
+      "grant_types": ["client_credentials"],
+    }'
+```
+
+</CoreInjector>
+
+
+- `client_name` - Then name of the client that will be used later for identification.
+- `grant_types` - The grant types that the [**Client**](/#authorization) will use. 
+    - `client_credentials`: Allows the client to directly request an Access Token by authenticating itself with the Authorization Server using its own client credentials.
+
+If the creation was successful, the response will return a `201` status code.
+The response body will contain a **`client_id`** field which you will need to use in the next steps.
+
+## 2. Initialize the OAuth2Provider recipe 
+
+
+In your **`authentication server`** you will need to initialize the **OAuth2Provider** recipe .
+The recipe will expose the endpoints needed for enabling the [**OAuth 2.0**](/docs/oauth/introduction#oauth-20) flow.
+
+<BackendSDKTabs>
+<TabItem value="nodejs">
+
+```typescript
+import supertokens from "supertokens-node";
+import OAuth2Provider from"supertokens-node/recipe/oauth2provider";
+import OAuth2Client from"supertokens-node/recipe/oauth2client";
+
+supertokens.init({
+    framework: "express",
+    supertokens: {
+        connectionURI: "<YOUR_CONNECTION_URI>",
+        apiKey: "<YOUR_API_KEY>",
+    },
+    appInfo: {
+        appName: "",
+        apiDomain: "<YOUR_API_DOMAIN>",
+        websiteDomain: "<YOUR_WEBSITE_DOMAIN>",
+        apiBasePath: "/auth",
+        websiteBasePath: "/auth"
+    },
+    recipeList: [
+        OAuth2Provider.init(),
+    ]
+});
+```
+
+</TabItem>
+
+<TabItem value="go">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Go SDK.
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
+
+
+## 3. Generate Access Tokens
+
+Now you can directly call the [**Authorization Server**](/docs/oauth/introduction#authorization-server) to generate Access Tokens.
+Check the following code snippet to see how you can do that:
+
+```bash
+
+curl -X POST <AUTHENTICATION_SERVICE_DOMAIN>/auth/oauth/token \
+-H "Content-Type: application/json" \
+-d '{
+  "client_id": "<YOUR_CLIENT_ID>",
+  "client_secret": "<YOUR_CLIENT_SECRET>",
+  "grant_type": "client_credentials",
+  "scope": ["<YOUR_REQUIRED_SCOPE>"]
+}'
+
+```
+
```

Comment:
Once the application gets the info from the above request, what then? How should they use the access token? For example, you may say that they should store those in memory, and reuse the access token for as long as its alive (how to check that it is alive?), and if not alive, they can generate a new access token.

How would the microservice getting the access token verify it (can link to a page that talks about oauth access token verification)? And once verified, it needs to check certain claims like the issuer for example?

Reply:
It should be stored in memory and used for some (fairly short IMHO) amount of time. 
They can see when it expires based on the response to the token request (it has an `expires_in` prop).
There are two ways of verifying:
- validating as a JWT
- querying the token introspection API